### PR TITLE
Lt 912 add an external note to a case

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: applications/views.py [hunk 1]
```diff
@@ -17,11 +17,12 @@ def index(request):
     return render(request, 'applications/index.html', context)
 
 
-def application(request, pk):
-    data, status_code = get_application(request, str(pk))
+def application(request, id):
+    data, status_code = get_application(request, str(id))
+    print(data)
```
Comment (cstobart) [inline applications/views.py]: remove print

File: ui_automation_tests/conftest.py [hunk 2]
```diff
@@ -41,10 +41,10 @@ def pytest_addoption(parser):
         env = "dev"
     print("touched: " + env)
```
Comment (cstobart) [inline ui_automation_tests/conftest.py]: remove print

File: ui_automation_tests/conftest.py [hunk 3]
```diff
@@ -41,10 +41,10 @@ def pytest_addoption(parser):
...
     parser.addoption("--driver", action="store", default="chrome", help="Type in browser type")
-    parser.addoption("--exporter_url", action="store", default="https://exporter.lite.service." + env + ".uktrade.io/", help="url")
-    parser.addoption("--internal_url", action="store", default="https://internal.lite.service." + env + ".uktrade.io/", help="url")
-    # parser.addoption("--exporter_url", action="store", default="localhost:8300/", help="url")
-    # parser.addoption("--internal_url", action="store", default="localhost:8200/", help="url")
+    # parser.addoption("--exporter_url", action="store", default="https://exporter.lite.service." + env + ".uktrade.io/", help="url")
```
Comment (cstobart) [inline ui_automation_tests/conftest.py]: reverse commenting so tests run against dev

File: ui_automation_tests/conftest.py (hunk 2 above)
Comment (5pence) [reply ui_automation_tests/conftest.py]: removed Nick's print

File: ui_automation_tests/conftest.py (hunk 3 above)
Comment (5pence) [reply ui_automation_tests/conftest.py]: done

File: applications/views.py (hunk 1 above)
Comment (5pence) [reply applications/views.py]: done

File: ui_automation_tests/conftest.py (hunk 3 above)
Comment (5pence) [reply ui_automation_tests/conftest.py]: good catch - thanks

File: Pipfile [hunk 4]
```diff
@@ -9,7 +9,7 @@ selenium = "==3.141.0"
 pytest = "==4.4.1"
 pytest_bdd = "==3.1.0"
 allure-pytest = "==2.6.1"
-pytest-html = "==1.21.1"
+pytest-html = "*"
```
Comment (efahy) [inline Pipfile]: Is there a reason you unpinned this version?  Was it to fix an issue you encountered?  If not, can you revert this change and if so can you pin to whatever new version fixes the issue please

File: ui_automation_tests/conftest.py [hunk 5]
```diff
@@ -15,17 +19,22 @@
...
 
 # from core import strings
 
 strict_gherkin = False
 
 
+sso_email = env('TEST_SSO_EMAIL')
+sso_password = env('TEST_SSO_PASSWORD')
+EXPORTER_URL = env('EXPORTER_FRONTEND_URL')
+INTERNAL_URL = env('INTERNAL_FRONTEND_URL')
```
Comment (efahy) [inline ui_automation_tests/conftest.py]: Nick says you can't put these URLs in the .env because of Jenkins complications

File: ui_automation_tests/conftest.py [hunk 6]
```diff
@@ -40,10 +49,8 @@ def pytest_addoption(parser):
     if env == 'None':
         env = "dev"
     parser.addoption("--driver", action="store", default="chrome", help="Type in browser type")
-    parser.addoption("--exporter_url", action="store", default="https://exporter.lite.service." + env + ".uktrade.io/", help="url")
-    parser.addoption("--internal_url", action="store", default="https://internal.lite.service." + env + ".uktrade.io/", help="url")
-    # parser.addoption("--exporter_url", action="store", default="localhost:8300/", help="url")
-    # parser.addoption("--internal_url", action="store", default="localhost:8200/", help="url")
+    parser.addoption("--exporter_url", action="store", default=EXPORTER_URL, help="url")
+    parser.addoption("--internal_url", action="store", default=INTERNAL_URL, help="url")
```
Comment (efahy) [inline ui_automation_tests/conftest.py]: Revert these to the previous versions on lines 43 & 44

File: ui_automation_tests/conftest.py [hunk 7]
```diff
@@ -390,3 +402,163 @@ def add_new_good(driver, description, controlled, control_code, incorporated, pa
...
+    application_page.enter_case_note(text)
+    application_page.click_visible_to_exporter_checkbox()
+    application_page.click_post_note_btn()
+    alert = driver.switch_to_alert()
+    alert.accept()
+
+
+
+@pytest.fixture(scope="module")
+def set_up_application_before_hook(driver, request, exporter_url, org_setup):
```
Comment (efahy) [inline ui_automation_tests/conftest.py]: Good work and thank you for getting these fixtures in place!

Comment (antonhughes) [reply ui_automation_tests/conftest.py]: Thank you :)

File: ui_automation_tests/conftest.py (hunk 6 above)
Comment (5pence) [reply ui_automation_tests/conftest.py]: reverted

File: ui_automation_tests/conftest.py (hunk 5 above)
Comment (5pence) [reply ui_automation_tests/conftest.py]: this is now sorted

**Comment by 5pence** [comment]
all changes complete and all tests passing
